runtime/src/master.rs

```diff
@@ -169,12 +169,15 @@ decl_module! {
         /// Execute a proposal that has received enough votes. The proposal is a serialized Call.
         /// This function can be called by anyone, even someone who is not a member of Master.
         ///
-        /// After a sucessful execution, the round number is increased.
+        /// After a successful execution, the round number is increased.
         // TODO: benchmark worst case cost to verify a signature and add it to weight
         #[
-            weight = 10_000
+            weight = (10_000
                 + proposal.get_dispatch_info().weight
-                + T::DbWeight::get().reads(auth.len() as u64)
+                + T::DbWeight::get().reads(auth.len() as u64),
+             proposal.get_dispatch_info().class,
+             proposal.get_dispatch_info().pays_fee,
+            )
         ]
         pub fn execute(
```

Comment:
This function will still charge for executing calls even when they return `Pays::No`. I don't think that's what we intend.

Reply:
What would cause the deduction?

Reply:
My understanding is that `proposal.get_dispatch_info().pays_fee` returns the pre-dispatch info. It can't run the extrinsic and check whether it returned `Pays::No` because that would be too expensive.

Master needs to either return `Pays::No` or propagate the return value from `proposal.dispatch()` if calls are to be free.

Reply:
Yes, you are correct. I missed modifying the return value of `execute`. It should return what `proposal`'s `dispatch` call returned.

Reply:
Fixed in https://github.com/docknetwork/dock-substrate/pull/76/commits/5923a238b11732bb41e40224288467f15e3706c6